test/controllers/api/accounts_controller_test.rb

```diff
@@ -140,4 +138,109 @@ class Api::AccountsControllerTest < ActionController::TestCase
       assert_equal 'password', account['cleartext_password']
     end
   end
+
+  context '#api_user' do
+    test 'authenticates with valid api user and returns account details' do
+      api_user.update!(valid_until: DateTime.now + 5.minutes)
+    
+      teams(:team1).add_user(api_user, plaintext_team_password)
+
+      request.env['HTTP_API_USER'] = api_user.username
+      request.env['HTTP_API_TOKEN'] = token
+
+      account = accounts(:account1)
+      get :show, params: { id: account }, xhr: true
+
+      account = JSON.parse(response.body)['data']['account']
+
+      assert_equal 'account1', account['accountname']
+      assert_equal 'test', account['cleartext_username']
+      assert_equal 'password', account['cleartext_password']
+    end
+    
+    test 'does not authenticate with invalid api token and does not show account details' do
+      api_user.update!(valid_until: DateTime.now + 5.minutes)
+    
+      teams(:team1).add_user(api_user, plaintext_team_password)
+
+      request.env['HTTP_API_USER'] = api_user.username
+      request.env['HTTP_API_TOKEN'] = 'abcd'
+
+      account = accounts(:account1)
+      get :show, params: { id: account }, xhr: true
+
+      assert_includes errors, 'Authentification failed'
+      assert_equal 401, JSON.parse(response.code)
+    end
+    
+    test 'cannot authenticate without headers and does not show account details' do
+      api_user.update!(valid_until: DateTime.now + 5.minutes)
+    
+      teams(:team1).add_user(api_user, plaintext_team_password)
+
+      account = accounts(:account1)
+      get :show, params: { id: account }, xhr: true
+
+      assert_redirected_to login_login_path
+      assert_nil response.body['data']
+    end
+    
+    test 'does not show account details if valid api user not teammember' do
+      api_user.update!(valid_until: DateTime.now + 5.minutes)
+    
+      request.env['HTTP_API_USER'] = api_user.username
+      request.env['HTTP_API_TOKEN'] = token
+
+      account = accounts(:account1)
+      get :show, params: { id: account }, xhr: true
+
+      assert_includes errors, 'Access denied'
+    end
+    
+    test 'cannot authenticate with api user with session' do
```

Comment:
1. login as bob
2. make request with headers
3. make sure api token authenticator is never called

you are not testing the right thing here